src/safeposix/syscalls/net_calls.rs

```diff
@@ -1234,4 +1233,274 @@ impl Cage {
+                    };
+                    ret_addr.set_addr(addr);
+                    ret_addr.set_port(0);
+                    ret_addr.set_family(sockfdobj.domain as u16);
+                    return 0;
+                }
+                
+                //if the socket is not none, then return the socket
+                *ret_addr = sockfdobj.localaddr.unwrap();
+                return 0;
+
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getsockname", "the provided file is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getsockname", "the provided file descriptor is not valid");
+        }
+    }
+
+    //we only return the default host name because we do not allow for the user to change the host name right now
+    pub fn gethostname(&self, length: usize, address_ptr: &mut [u8]) -> i32 {
+        if length < 0 {
+            return syscall_error(Errno::EINVAL, "gethostname", "invalid argument");
+        }
+
+        let name_length: usize = DEFAULT_HOSTNAME.chars().count();
+        if name_length > length {
+            address_ptr[..length].copy_from_slice(&DEFAULT_HOSTNAME[..length].as_bytes());
+        } else {
+            address_ptr[..name_length].copy_from_slice(&DEFAULT_HOSTNAME.as_bytes());
+        }
+        return 0;
+    }
+
+    pub fn poll_syscall(&self, fds: &mut [PollStruct], timeout: Option<interface::RustDuration>) -> i32 { //timeout is supposed to be in milliseconds
+
+        let mut return_code: i32 = 0;
+        let start_time = interface::starttimer();
+
+        let end_time = match timeout {
+            Some(time) => time,
+            None => interface::RustDuration::MAX
+        };
+
+        loop {
+            for structpoll in &mut *fds {
+                let fd = structpoll.fd;
+                let events = structpoll.events;
+
+                let read = if events & POLLIN > 0 {true} else {false};
+                let write = if events & POLLOUT > 0 {true} else {false};
+                let err = if events & POLLERR > 0 {true} else {false};
+
+                let mut reads = interface::RustHashSet::<i32>::new();
+                let mut writes = interface::RustHashSet::<i32>::new();
+                let mut errors = interface::RustHashSet::<i32>::new();
+
+                let mut mask: u32 = 0;
+
+                //passing None does the same thing as passing 0 in native Linux
+                //which essentially sets the timeout to the max value allowed (which is almost always more than enough time)
+                if Self::select_syscall(&self, fd, &mut reads, &mut writes, &mut errors, None) > 0 {
+                    mask += if !reads.is_empty() {POLLIN} else {0};
+                    mask += if !writes.is_empty() {POLLOUT} else {0};
+                    mask += if !errors.is_empty() {POLLERR} else {0};
+                    return_code += 1;
+                }
+                structpoll.revents = mask;
```

Comment:
You need to put the fds you want to look at in the sets. What you do is set a boolean flag if it's POLL_IN etc. but you don't actually put it into the requisite set in that case, which would be necessary for select to return anything.

Reply:
Furthermore, just a general comment about programming style, you really pretty much never want an if statement that yields/assigns true and false in the if/else cases, because in all such cases the if statement is unnecessary:
i.e.
`if expression {return true;} else {return false;}` can always be replaced by `return expression;`